swupd/manifest.go

```diff
@@ -439,9 +439,12 @@ func (m *Manifest) linkPeersAndChange(oldManifest *Manifest, minVersion uint32)
 		added = append(added, nf)
 	}
 
-	// anything remaining in oldManifest is newly deleted in the new manifest
+	// anything remaining in oldManifest is deleted in the new manifest
 	for _, of := range oldManifest.Files[ox:omFilesLen] {
-		if of.Status == StatusDeleted || of.Status == StatusGhosted {
```

Comment:
@otaviobp @phmccarty Are either of you familiar with the expected behavior of ghosted files?

Currently, the **behavior of ghosted files is inconsistent**. Based on the name value of the ghosted file (different logic is used when the name is at the beginning or end of the sorted file list), the following scenarios can happen:
1. The ghosted file is dropped from the new manifest
2. The ghosted file is considered a newly deleted file in the new manifest

This change makes ghosted files follow `scenario 1` which is the most common case.

I am not familiar with ghosted files, but I don't feel like dropping them or marking them as newly deleted files is correct. Does it make sense for ghosted files to persist in future versions of the manifest within the same format?

Reply:
IIRC, ghosted files should only appear in exactly one version of a manifest. They are supposed to disappear from the manifests in the next build that the manifest changes.

Reply:
For more background: ghosted files were implemented for the kernel-* bundles so that mixer could create (rename) deltas for kernel modules/images.

Reply:
Thanks for the feedback. In that case, this PR also fixes the behavior of ghosted files. When a manifest with ghosted files is updated, the ghosted file entries will be dropped in the new manifest.

Reply:
Great! Sounds good to me.